libraries/core_libs/consensus/src/dag/sortition_params_manager.cpp

```diff
@@ -74,30 +49,32 @@ SortitionParamsManager::SortitionParamsManager(const addr_t& node_addr, Sortitio
     config_.vrf = params_changes_.back().vrf_params;
   }
 
-  dag_efficiencies_ = db_->getLastIntervalEfficiencies(config_.changing_interval, config_.computation_interval);
+  auto period = params_changes_.back().period + 1;
+  ignored_efficiency_counter_ = 0;
+  while (true) {
+    auto data = db_->getPeriodDataRaw(period);
+    if (data.size() == 0) break;
+    period++;
+    SyncBlock sync_block(data);
+    if (sync_block.pbft_blk->getPivotDagBlockHash() != NULL_BLOCK_HASH) {
+      if (ignored_efficiency_counter_ >= config_.computation_interval - config_.changing_interval) {
+        uint16_t dag_efficiency = calculateDagEfficiency(sync_block);
+        dag_efficiencies_.push_back(dag_efficiency);
+      } else {
+        ignored_efficiency_counter_++;
+      }
+    }
+  }
 }
```

Comment:
So we save params_change every 200 non empty pbft blocks but only do calculation on last 50 blocks, so on startup this will do up to max 50 non empty pbft blocks. From my testing this is really fast, usually it runs under one second.